src/lightning/data/datasets/env.py

```diff
@@ -0,0 +1,147 @@
+from typing import Optional
+
+import torch
+from torch.utils.data import get_worker_info
+
+
+class DistributedEnv:
```

Comment:
I suggest making these helper classes protected. I expect they need to be changed in the future, so it would be easier to do if we didn't expose them to users (yet). wdyt?